executor/prepared.go

```diff
@@ -341,10 +341,18 @@ func ResetStmtCtx(ctx context.Context, s ast.StmtNode) {
 		sc.OverflowAsWarning = false
 		sc.TruncateAsWarning = !sessVars.StrictSQLMode
 		sc.InUpdateOrDeleteStmt = true
+		if node, ok := s.(*ast.UpdateStmt); ok {
+			sc.IgnoreError = node.IgnoreErr
```

Comment:
use stmt.IgnoreErr is enough, the type assertion in line 344 can be removed.

Reply:
If I use `stmt.IgnoreErr`, the compile will fail(`stmt.IgnoreErr undefined (type ast.StmtNode has no field or method IgnoreErr)` ). However `ast.StmtNode` is an interface. So any better way to do the same effect?

Reply:
seperate line 339 will make it ok.